sfge/src/test/java/com/salesforce/graph/ops/MethodUtilTest.java

```diff
@@ -0,0 +1,282 @@
+        MethodVertex firstVertex = methodVertices.get(0);
+        assertEquals(METHOD_WITH_INTERNAL_OVERLOADS, firstVertex.getName());
+
+        MethodVertex secondVertex = methodVertices.get(1);
+        assertEquals(METHOD_WITH_INTERNAL_OVERLOADS, secondVertex.getName());
+
+        String messages = CliMessager.getInstance().getAllMessages();
+        MatcherAssert.assertThat(
+                messages,
+                containsString(EventKey.WARNING_MULTIPLE_METHOD_TARGET_MATCHES.getMessageKey()));
+    }
+
+    @Test
+    public void getTargetMethods_targetNameDupedMethods() {
+        TestUtil.Config config =
+                TestUtil.Config.Builder.get(g, new String[] {SOURCE_FILE_1, SOURCE_FILE_2}).build();
+        TestUtil.buildGraph(config);
+        // Create a rule target encompassing only the method in the first file whose name is used
+        // elsewhere.
+        List<RuleRunnerTarget> targets = new ArrayList<>();
+        targets.add(
+                new RuleRunnerTarget(
+                        "TestCode0",
+                        Collections.singletonList(METHOD_WITH_EXTERNAL_NAME_DUPLICATION)));
+
+        List<MethodVertex> methodVertices = MethodUtil.getTargetedMethods(g, targets);
+
+        MatcherAssert.assertThat(methodVertices, hasSize(equalTo(2)));
+        MethodVertex firstVertex = methodVertices.get(0);
+        assertEquals(METHOD_WITH_EXTERNAL_NAME_DUPLICATION, firstVertex.getName());
+        assertEquals(18, firstVertex.getBeginLine());
+
+        MethodVertex secondVertex = methodVertices.get(1);
+        assertEquals(METHOD_WITH_EXTERNAL_NAME_DUPLICATION, secondVertex.getName());
+        assertEquals(22, secondVertex.getBeginLine());
+
+        String messages = CliMessager.getInstance().getAllMessages();
+        MatcherAssert.assertThat(
+                messages,
+                containsString(EventKey.WARNING_MULTIPLE_METHOD_TARGET_MATCHES.getMessageKey()));
+    }
+
+    @Test
+    public void getTargetMethods_targetMethodDoesNotExist() {
+        TestUtil.Config config =
+                TestUtil.Config.Builder.get(g, new String[] {SOURCE_FILE_1, SOURCE_FILE_2}).build();
+        TestUtil.buildGraph(config);
+        // Create a rule target encompassing a method that doesn't actually exist with that name.
+        List<RuleRunnerTarget> targets = new ArrayList<>();
+        targets.add(
+                new RuleRunnerTarget(
+                        "TestCode0", Collections.singletonList(METHOD_THAT_DOES_NOT_EXIST)));
+
+        List<MethodVertex> methodVertices = MethodUtil.getTargetedMethods(g, targets);
+
+        MatcherAssert.assertThat(methodVertices, hasSize(equalTo(0)));
+
+        String messages = CliMessager.getInstance().getAllMessages();
+        MatcherAssert.assertThat(
+                messages,
+                containsString(EventKey.WARNING_NO_METHOD_TARGET_MATCHES.getMessageKey()));
+    }
+
+	@Test
+	public void getTargetMethods_targetMethodInInnerClass() {
```

Comment:
Test case: Targeting a method in an inner class. As noted below, the syntax would be `path/to/file.cls#methodName`, without including the inner class.